src/ipc/ipc.c

```diff
@@ -309,10 +311,20 @@ int ipc_buffer_free(struct ipc *ipc, uint32_t buffer_id)
 	if (!cpu_is_me(ibd->core))
 		return ipc_process_on_core(ibd->core);
 
-	/* check state */
-	if (ibd->cb->sink->state != COMP_STATE_READY ||
-	    ibd->cb->source->state != COMP_STATE_READY)
-		return -EINVAL;
+	/* try to find sink/source components */
```

Comment:
this is better, but I want to know why we are doing this check. What condition are we checking is true before we free ?

Reply:
Updated, note that it is divided into 2 comments (another one inside this loop)